src/components/Dialogs/DimensionModal.js

```diff
@@ -0,0 +1,68 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    Modal,
+    ModalContent,
+    ModalActions,
+    ButtonStrip,
+    ModalTitle,
+    Button,
+} from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { AddToLayoutButton } from './AddToLayoutButton/AddToLayoutButton'
+
+const DimensionModal = ({
+    content,
+    dataTest,
+    isInLayout,
+    onClose,
+    onUpdate,
+    title,
+}) => {
+    return (
+        <Modal onClose={onClose} dataTest={`${dataTest}`} position="top" large>
+            <ModalTitle dataTest={`${dataTest}-title}`}>{title}</ModalTitle>
+            <ModalContent dataTest={`${dataTest}-content`}>
+                {content}
```

Comment:
I think we should consider using the `children` prop for populating the modal content. This will enable us to move away from the `renderModalContent` helpers and simply insert them into the `DimensionsModal`.

Reply:
Good point! Refactoring this now!